packages/evm/src/evmTransfer.ts

```diff
@@ -1,31 +1,46 @@
-import type { BaseTransferParams, Config, Eip1193Provider } from '@buildwithsygma/core';
+import type { Config, HexString, Eip1193Provider } from '@buildwithsygma/core';
 import { BaseTransfer } from '@buildwithsygma/core';
 import { providers } from 'ethers';
 
 import { TwapFeeCalculator } from './fee/TwapFee.js';
 import { getFeeInformation, BasicFeeCalculator, PercentageFeeCalculator } from './fee/index.js';
-import type { EvmFee } from './types.js';
+import type { EvmFee, EvmTransferParams } from './types.js';
 
-export interface EvmTransferParams extends BaseTransferParams {
-  sourceNetworkProvider: Eip1193Provider;
-}
+/**
+ * @internal
+ * @class EvmTransfer
+ *
+ * @abstract
+ * Base EVM transfer class
+ * housing common functionality
+ */
+export abstract class EvmTransfer extends BaseTransfer {
+  protected provider: Eip1193Provider;
 
-export class EvmTransfer extends BaseTransfer {
-  sourceNetworkProvider: Eip1193Provider;
+  get sourceNetworkProvider(): Eip1193Provider {
+    return this.provider;
+  }
 
-  constructor(params: EvmTransferParams, config: Config) {
+  protected constructor(params: EvmTransferParams, config: Config) {
```

Comment:
should all the subclasses then have protected constructor?

Reply:
@wainola only abstract classes

Reply:
solved